lib/Parse/Parser.cpp

```diff
@@ -1185,6 +1185,29 @@ Parser::getStringLiteralIfNotInterpolated(SourceLoc Loc,
                                                Segments.front().Length));
 }
 
+bool Parser::shouldReturnSingleExpressionElement(ArrayRef<ASTNode> Body) {
+  // If the body consists of an #if declaration with a single
+  // expression active clause, find a single expression.
+  if (Body.size() == 2) {
+    if (auto *D = Body.front().dyn_cast<Decl *>()) {
+      // Step into nested active clause.
+      while (auto *ICD = dyn_cast<IfConfigDecl>(D)) {
+        auto ACE = ICD->getActiveClauseElements();
+        if (ACE.size() == 1) {
+          return true;
+        } else if (ACE.size() == 2) {
+          if (auto *ND = ACE.front().dyn_cast<Decl *>()) {
+            D = ND;
+            continue;
+          }
+        }
+        break;
+      }
+    }
```

Comment:
I think this can't handle
```swift
func foo() -> Int {
#if true
  #if true
  #endif
  1
#endif
}
```
Is it intentional? Also, how about
```swift
func foo() -> Int {
#if false
  ...
#endif
  1
#if false
  ...
#endif
}
```

Reply:
This was intentional, but I am happy to change it if you think the behavior makes sense.

I would be hesitant to change this because:
- Certain flag combinations could result in invalid code.
- Any flag state can already be represented within a single #if declaration

My reasoning is that, most likely, a compiler flag would be used instead of a literal true or false. And for the cases you mention, I see how the flags can align into a cases that appear valid, but also allow for invalid code with different flag values.

I did not think to support either case because it allows for states with an unused literal when content is added within the nested #if declaration, and possibly an error for missing a return statement.

This alternative implementation would search all elements, step into all #if declarations, find one active expression to return, and quit if a second expression is found. Do you think that behavior is expected as opposed to the current method?

Reply:
OK. I understand, and I can agree that such `#if` usages are not "single expression".